example/src/main/java/com/mattskala/itemadapter/example/MainActivity.kt

```diff
@@ -42,29 +48,36 @@ class MainActivity : AppCompatActivity() {
         when (item.itemId) {
             ITEM_ADD -> {
                 val items = adapter.items.toMutableList()
-                items.add(ExampleItem("What Next?", "X.Y"))
+                items.add(ExampleItem(items.size, "What Next?", "X.Y", false))
```

Comment:
You forgot to update the local `items` value here, so when you add a new item and try to check it, it gets removed.